src/hostapi/asio/pa_asio.cpp

```diff
@@ -1871,24 +1874,43 @@ static PaError ValidateAsioSpecificStreamInfo(
 {
     if( streamInfo )
     {
-        if( streamInfo->size != sizeof( PaAsioStreamInfo )
-                || streamInfo->version != 1 )
+        switch( streamInfo->version )
         {
+        case 1:
+            /* NOTE: V1 structure's size is smaller by one pointer. */
+            if( streamInfo->size < sizeof( PaAsioStreamInfo ) - sizeof(PaAsio_MessageCallback*) )
+                return paIncompatibleHostApiSpecificStreamInfo;
+            break;
+            
+        case 2:
+            if( streamInfo->size != sizeof( PaAsioStreamInfo ) )
+                return paIncompatibleHostApiSpecificStreamInfo;
+            break;
+            
+        default:
             return paIncompatibleHostApiSpecificStreamInfo;
         }
 
         if( streamInfo->flags & paAsioUseChannelSelectors )
```

Comment:
It looks like you've fixed a bug here by making all channel selector handling depend on the condition `streamInfo->flags & paAsioUseChannelSelectors`. This seems like a valid fix. Could you extract this out to a separate PR please? We can merge that immediately.

Reply:
I did not change the behavior of the paAsioUseChannelSelectors flag.  That condition was there before.

Reply:
You did change the behavior: you added braces on lines 1902-1920 so the scope of the conditional is changed.

Reply:
OK, I've created PR https://github.com/PortAudio/portaudio/pull/885 comprising just this fix and removed it from here.  We'll want to merge 885 first.